spark/src/main/scala/org/apache/spark/sql/delta/actions/TableFeatureSupport.scala

```diff
@@ -258,24 +269,32 @@ trait TableFeatureSupport { this: Protocol =>
   def canDowngradeTo(to: Protocol, droppedFeatureName: String): Boolean = {
     if (!supportsWriterFeatures) return false
 
-    // When `to` protocol does not have any features version downgrades are possible. However,
+    val featureNames = readerAndWriterFeatureNames - droppedFeatureName
+    val minRequiredVersions = TableFeatureProtocolUtils.minimumRequiredVersions(
+      featureNames.flatMap(TableFeature.featureNameToFeature).toSeq)
+    val toSupportsRequiredMinVersions =
+      (to.minReaderVersion, to.minWriterVersion) == minRequiredVersions
```

Comment:
I wonder can we compare if
```
Protocol(minRequiredVersions._1, minRequiredVersions._2).implicitlyAndExplicitlySupportedFeatures ===
   to.implicitlyAndExplicitlySupportedFeatures
```
instead?
Only when they supports the same feature we could downgrade.

Reply:
In the case of version downgrade with table features that would not work. The issue is that `Protocol(minRequiredVersions._1, minRequiredVersions._2).implicitlyAndExplicitlySupportedFeatures` would have an empty explicit features set. This is because `implicitlyAndExplicitlySupportedFeatures` generates only the implicit features while the the explicit feature have to be added, well, explicitly :D.

Reply:
I see!